compose/service.py

```diff
@@ -448,11 +479,26 @@ def recreate_container(self,
             else:
                 raise
 
+    def _recreate_rename_container(self, container):
         # Use a hopefully unique container name by prepending the short id
         self.client.rename(
             container.id,
-            '%s_%s' % (container.short_id, container.name))
+            '%s_%s' % (container.short_id, container.name)
+        )
+
+    def recreate_container(self,
+                           container,
+                           timeout=DEFAULT_TIMEOUT):
```

Comment:
Instead of duplicating this, with my proposed change to add init_log_stream/get_log_stream to Container, I think we can just add an `init_log_stream=False` kwarg here, and set it to `True` from `execute_convergence_plan()` to remove the need for duplication.